components/ListServices/mint.jsx

```diff
@@ -16,26 +16,80 @@ import { getServiceManagerContract } from 'common-util/Contracts';
 import { sendTransaction } from 'common-util/functions';
 import { checkIfERC721Receive } from 'common-util/functions/requests';
 import { useHelpers } from 'common-util/hooks';
+import { useSvmInfo } from 'common-util/hooks/useSvmInfo';
+import { BN } from '@project-serum/anchor';
 import RegisterForm from './RegisterForm';
 import { getAgentParams } from './utils';
 import { FormContainer } from '../styles';
 
 const { Title } = Typography;
 
 const MintService = () => {
-  const router = useRouter();
-  const { account, chainName, doesNetworkHaveValidServiceManagerToken } = useHelpers();
+  const {
+    account, doesNetworkHaveValidServiceManagerToken, vmType, isSvm,
+  } = useHelpers();
+  const { getAddresses } = useSvmInfo();
 
   const [isMinting, setIsMinting] = useState(false);
   const [error, setError] = useState(null);
   const [information, setInformation] = useState(null);
 
+  const { wallet, program } = useSvmInfo();
+
+  const buildSvmFn = async (values) => {
+    const {
+      owner_address: ownerAddress,
+      hash,
+      agent_ids: agentIdsSrc,
+      agent_num_slots: slotsSrc,
+      bonds,
+      threshold: thresholdStr,
+    } = values;
+
+    const serviceOwnerPublicKey = ownerAddress;
```

Comment:
imo it's worth this for clarity because this maps closely to the IDL:

<img width="384" alt="image" src="https://github.com/valory-xyz/autonolas-registry-frontend/assets/66292936/390f52c1-c520-446d-940b-03aeb12ffc99">